src/core/data.test.js

```diff
@@ -0,0 +1,222 @@
+  });
+
+  test("instanciate the data with default options (null)", () => {
+    const data = new Data();
+    const expectedOptions = {
+      startSymbol: "{{",
+      endSymbol: "}}"
+    };
+    expect(data.options).toEqual(expectedOptions);
+  });
+
+  test("instanciate the data using a different symbols", () => {
+    const options = {
+      startSymbol: "{[",
+      endSymbol: "]}"
+    };
+    const data = new Data(options);
+    const expectedOptions = {
+      startSymbol: "{[",
+      endSymbol: "]}"
+    };
+    expect(data.options).toEqual(expectedOptions);
+  });
+
+  test("Throw an error when trying to instanciate if the startSymbole is that more than characters", () => {
+    expect(() => {
+      const options = {
+        startSymbol: "{[[",
+        endSymbol: "]}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The startSymbol should contains 2 charaters");
+
+    expect(() => {
+      const options = {
+        startSymbol: "[",
+        endSymbol: "]}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The startSymbol should contains 2 charaters");
+  });
+
+  test("Throw an error when trying to instanciate if the endSymbole is that more than characters", () => {
+    expect(() => {
+      const options = {
+        startSymbol: "{[",
+        endSymbol: "]}}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The endSymbol should contains 2 charaters");
+
+    expect(() => {
+      const options = {
+        startSymbol: "{[",
+        endSymbol: "]"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The endSymbol should contains 2 charaters");
+  });
+
+  test("add/get values to the datastore (string)", () => {
+    const options = {};
+    const data = new Data(options);
+    data.set("foo", "bar");
+    expect(data.get("foo")).toEqual("foo");
+    expect(data.get("{{foo")).toEqual("{{foo");
+    expect(data.get("{{ foo}}")).toEqual("bar");
+    expect(data.get("{{ foo }}")).toEqual("bar");
+    expect(data.get("{{foo }}")).toEqual("bar");
+    expect(data.get("{{foo}}")).toEqual("bar");
+    expect(data.get("{{foo o }}")).toEqual("{{foo o }}");
+  });
+
+  test("add/get values to the datastore (integer)", () => {
+    const options = {};
+    const data = new Data(options);
+    data.set("foo", 2021);
+    expect(data.get("foo")).toEqual("foo");
+    expect(data.get("{{foo")).toEqual("{{foo");
+    expect(data.get("{{ foo}}")).toEqual(2021);
+    expect(data.get("{{ foo }}")).toEqual(2021);
+    expect(data.get("{{foo }}")).toEqual(2021);
+    expect(data.get("{{foo}}")).toEqual(2021);
+    expect(data.get("{{foo}} year")).toEqual("2021 year");
+    expect(data.get("{{foo o }} Nice")).toEqual("{{foo o }} Nice");
+  });
+
+  test("Do not parse the data from the scenario since the there is no channel associated", () => {
```

Comment:
Is no channel related to no options?

Reply:
I think it's more something like `{ channel: undefined }` that we need here. It's add a bit of expressiveness.